google/cloud/spanner/database_admin_client.h

```diff
@@ -226,14 +230,17 @@ class DatabaseAdminClient {
    * The new database must be in the same project and in an instance with the
    * same instance configuration as the instance containing the backup.
    *
+   * @p encryption_config How to encrypt the database.
+   *
    * @return A `google::cloud::future` that becomes satisfied when the operation
    *   completes on the service. Note that this can take minutes in some cases.
    *
    * @par Example
    * @snippet samples.cc restore-database
    */
   future<StatusOr<google::spanner::admin::database::v1::Database>>
-  RestoreDatabase(Database db, Backup const& backup);
+  RestoreDatabase(Database db, Backup const& backup,
+                  EncryptionConfig encryption_config = DefaultEncryption());
```

Comment:
You probably know this and decided otherwise, but I'll say that default constructing a variant constructs a value-initialized instance of its first member, which is a `DefaultEncryption`. It's good that you arranged things this way. That way, `EncryptionConfig x = {}`, does the right thing and is equivalent to `EncryptionConfig x = DefaultEncryption{}`. So these arguments could be written either way. I'll let you choose the one you prefer. I'm happy either way.

Reply:
Yes ... I thought this way made it clearer to the peruser of the declaration exactly which behavior they would get in the defaulted case.